accounts/abi/bind/bind.go

```diff
@@ -650,6 +650,11 @@ func decapitalise(input string) string {
 func convertToNil(input abi.Type) string {
 	switch input.T {
 	case abi.IntTy, abi.UintTy:
+		parts := regexp.MustCompile(`(u)?int([0-9]*)`).FindStringSubmatch(input.String())
+		switch parts[2] {
+		case "8", "16", "32", "64":
+			return "0"
+		}
```

Comment:
Can we add unit tests for `convertToNil` ?

Reply:
If we need a regex here, should we compile once with `MustCompile` and re-use it rather than compiling every time this is called?

Reply:
I think using `*new(Type)` would be a more straightforward solution. I removed all of these and converted template to use that instead.

Reply:
Won't this generate some potentially weird code like `*new(int)` when we really just want 0?

Reply:
Yes but `convertToNil` cannot correctly detect array types + custom structs. We can expand the function by copying the`bindTypeGo`, but seemed a bit too much copy pasta to me.

This is also being used by ABI binder: https://github.com/ava-labs/subnet-evm/blob/master/accounts/abi/bind/template.go#L344